cpp/src/gandiva/function_ir_builder.cc

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gandiva/function_ir_builder.h"
+
+namespace gandiva {
+
+llvm::Value* FunctionIRBuilder::BuildIfElse(llvm::Value* condition,
```

Comment:
Isn't this function just reimplementing the `select` instruction? https://llvm.org/docs/LangRef.html#select-instruction

Reply:
no - select-instruction is only conditional assignment. It doesn't do control flow.

For the same example as above, with select (instead of if-else), I get 

  ```
%2 = icmp slt i32 %member1, 38
  %3 = add i128 %member, 11
  %4 = sub i128 %member, 11
  %5 = select i1 %2, i128 %3, i128 %4
```